### PR TITLE
(#1223) Replace static factory methods with direct call to new

### PR DESCRIPTION
**Pull request by andreoss**
Per #1223: 
- Replace static methods with objects
- Update puzzle

### CODE REVIEW

**Comment by victornoel** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/cactoos/issues/1433#issuecomment-686931360))

@victornoel This is what I know about this job in C63314D6Z, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1433](https://github.com/yegor256/cactoos/issues/1433) is [in scope](http://www.zerocracy.com/policy.html#14) for 2days
  * The role is `REV`
  * The job is not assigned to anyone
  * The [budget](http://www.zerocracy.com/policy.html#4) is not set yet
  * These users are banned and won't be assigned:
    * @andreoss[/z](https://www.0crat.com/u/andreoss): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C63314D6Z?q=%7Bjob%3A%27gh%3Ayegor256%2Fcactoos%231433%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by victornoel** [comment]
@0crat assign me

File: src/test/java/org/cactoos/func/UncheckedBiFuncTest.java [hunk 1]
```diff
@@ -52,7 +52,7 @@ public void testUncheckedBiFunc() {
             new UncheckedBiFunc<>(
                 (fst, scd) -> true
             ).apply(1, 2),
-            Matchers.equalTo(true)
+            new IsEqual<>(true)
```
Comment (victornoel) [inline src/test/java/org/cactoos/func/UncheckedBiFuncTest.java:55]: @andreoss let's use `IsTrue` from cactoos-matchers here :)

File: src/test/java/org/cactoos/func/TimedFuncTest.java [hunk 2]
```diff
@@ -88,7 +88,7 @@ public void functionIsExecuted() throws Exception {
                 input -> true,
                 period
             ).apply(true),
-            Matchers.equalTo(true)
+            new IsEqual<>(true)
```
Comment (victornoel) [inline src/test/java/org/cactoos/func/TimedFuncTest.java:91]: @andreoss let's use `IsTrue` from cactoos-matchers here :)

File: src/test/java/org/cactoos/func/TimedFuncTest.java [hunk 3]
```diff
@@ -74,7 +74,7 @@ public void futureTaskIsCancelled() {
             new Assertion<>(
                 "Must be canceled after 1 sec",
                 future.isCancelled(),
-                Matchers.equalTo(true)
+                new IsEqual<>(true)
```
Comment (victornoel) [inline src/test/java/org/cactoos/func/TimedFuncTest.java:77]: @andreoss let's use `IsTrue` from cactoos-matchers here :)

**Review by victornoel — CHANGES_REQUESTED** [review]
@andreoss a few comments :)

File: src/test/java/org/cactoos/func/TimedFuncTest.java (hunk 3 above)
Comment (andreoss) [reply src/test/java/org/cactoos/func/TimedFuncTest.java]: @victornoel Done

File: src/test/java/org/cactoos/func/TimedFuncTest.java (hunk 2 above)
Comment (andreoss) [reply src/test/java/org/cactoos/func/TimedFuncTest.java]: @victornoel Done

**Review by victornoel — APPROVED** [review]
@andreoss thank you :)

**Comment by victornoel** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 2 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Pull request merged**